kumascript/src/api/mdn.js

```diff
@@ -134,4 +143,42 @@ module.exports = {
       "This macro has been deprecated, and should be removed."
     );
   },
+
+  async fetchWebExtExamples() {
+    if (!webExtExamplesCached) {
+      try {
+        webExtExamplesCached = await got(
+          "https://raw.githubusercontent.com/mdn/webextensions-examples/master/examples.json",
+          {
+            timeout: 10000,
+          }
+        ).json();
+        if (process.env.NODE_ENV === "development") {
```

Comment:
First of all, `process.env.NODE_ENV` is `undefined` inside macros execution. 
So it's never cached. 

I don't like that there's even a distinction between prod and development because it's too different. I think it's fine to leverage `process.env.NODE_ENV` for things that aren't optimizations but stuff that actually has a functional difference. For example, we don't include the Flaws Dashboard or the Toollbar into the client bundle when we build it for production. 

The other thing I don't like is that we're using the repo's root as the cache. Using `/tmp` is much better. Then the OS will make sure the file doesn't get horribly outdated for most people. It also is harmless to use it for those `yarn build` production builds since it's always going to be a temp directory.

Reply:
> First of all, `process.env.NODE_ENV` is `undefined` inside macros execution.

Did you try it? The call to `process.env.NODE_ENV` here works just fine when called from within the macros.

> So it's never cached.

I just want to be clear here. The caching controlled here by `if (process.env.NODE_ENV === "development")` is only for saving the results in a file for **possible** future use when developing **offline**. It's not the caching of the result for use by later calls to `fetchWebExtExamples()`. That's cached in the `webExtExamplesCached` variable.

> I don't like that there's even a distinction between prod and development because it's too different. I think it's fine to leverage `process.env.NODE_ENV` for things that aren't optimizations but stuff that actually has a functional difference. For example, we don't include the Flaws Dashboard or the Toollbar into the client bundle when we build it for production.

The only distinction between prod and development here is when working offline. Both prod and development **always** try to get the latest web-ext examples via the network the first time `fetchWebExtExamples()` is called, but if that fails, only in development mode does it make the additional attempt to use the last result stored in the file. If we'd rather that the web-ext-example macros fail (with a macro flaw) in development mode just as they would in production when the network is unavailable, that's fine with me. It's much simpler and probably better.

> The other thing I don't like is that we're using the repo's root as the cache. Using `/tmp` is much better. Then the OS will make sure the file doesn't get horribly outdated for most people. It also is harmless to use it for those `yarn build` production builds since it's always going to be a temp directory.

If we decide to allow an offline mode, I like the idea of using `tempy`.